### PR TITLE
refactor: NSSF app, sbi, processor, consumer

### PR DESCRIPTION
**Pull request by yccodr**
# Description 
- Refactor: Nssf App, SBI Server, Processor, Consumer

# TODO

- [x] Test UE connection

### CODE REVIEW

**Comment by andy89923** [comment]
I think you forgot `NewRouterAuthorizationCheck`.

**Review by andy89923 — DISMISSED** [review]

File: pkg/service/init.go [hunk 1]
```diff
@@ -161,6 +110,41 @@ func (nssf *NssfApp) Terminate() {
 	} else {
 		logger.InitLog.Infof("Deregister from NRF successfully")
 	}
+}
+
+func (a *NssfApp) Start(ctx context.Context) {
```
Comment (andy89923) [inline pkg/service/init.go]: start(tlsKeyLogPath string)

Comment (andy89923) [reply pkg/service/init.go]: don't pass context here.

File: internal/context/context.go [hunk 2]
```diff
@@ -66,6 +66,8 @@ type NSSFContext struct {
 
 // Initialize NSSF context with configuration factory
 func InitNssfContext() {
+	Init()
```
Comment (ianchen0119) [inline internal/context/context.go]: Remove this line.

File: internal/context/context.go [hunk 3]
```diff
@@ -66,6 +66,8 @@ type NSSFContext struct {
...
+
```
Comment (ianchen0119) [inline internal/context/context.go]: add:
```
nssfContext.NfId = uuid.New().String()
nssfContext.Name = "NSSF"
```
and remove the `Init()` function.

File: internal/util/problem_json.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"github.com/gin-gonic/gin"
+
+	"github.com/free5gc/openapi/models"
+)
+
+func GinProblemJson(c *gin.Context, problemDetails *models.ProblemDetails) {
+	c.JSON(int(problemDetails.Status), problemDetails)
+	c.Writer.Header().Set("Content-Type", "application/problem+json")
```
Comment (ianchen0119) [inline internal/util/problem_json.go:11]: Should `c.Writer.Header().Set()` performs before the `c.JSON()`?
Please help to check, tks!

**Comment by ianchen0119** [comment]
Overall, it looks good to me.

File: internal/util/problem_json.go (hunk 4 above)
Comment (yccodr) [reply internal/util/problem_json.go:11]: Yes. `Header().set()` will replace the existing value, which is `Content-Type` in this case. We want to replace `Content-Type: application/json` with `Content-Type: application/problem+json` to align with api spec of r17.

See more details [here](https://pkg.go.dev/net/http#Header.Set)

File: internal/context/context.go (hunk 3 above)
Comment (yccodr) [reply internal/context/context.go]: @ianchen0119 Done!

**Review by ianchen0119 — APPROVED** [review]

File: pkg/service/init.go [hunk 5]
```diff
@@ -45,9 +45,6 @@ func NewApp(cfg *factory.Config, tlsKeyLogPath string) (*NssfApp, error) {
 	nssf.SetLogLevel(cfg.GetLogLevel())
 	nssf.SetReportCaller(cfg.GetLogReportCaller())
 
-	processor := processor.NewProcessor(nssf)
```
Comment (ianchen0119) [inline pkg/service/init.go]: add this back.

File: internal/sbi/server.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+	httpServer *http.Server
+	router     *gin.Engine
+	processor  *processor.Processor
+}
+
+func NewServer(nssf app.NssfApp, tlsKeyLogPath string) *Server {
+	s := &Server{
+		NssfApp:   nssf,
+		processor: processor.NewProcessor(nssf),
```
Comment (ianchen0119) [inline internal/sbi/server.go]: processor: nssf.Processor(),

File: pkg/service/init.go (hunk 5 above)
Comment (yccodr) [reply pkg/service/init.go]: @ianchen0119 Done!

File: internal/sbi/consumer/nrf_service.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+	nssf_context "github.com/free5gc/nssf/internal/context"
+	"github.com/free5gc/nssf/internal/logger"
+	"github.com/free5gc/openapi"
+	"github.com/free5gc/openapi/Nnrf_NFManagement"
+	"github.com/free5gc/openapi/models"
+)
+
+type NrfService struct {
+	nrfNfMgmtClient *Nnrf_NFManagement.APIClient
```
Comment (andy89923) [inline internal/sbi/consumer/nrf_service.go:24]: Add mutex 
```go
nfMngmntMu sync.RWMutex
```

Comment (yccodr) [reply internal/sbi/consumer/nrf_service.go:24]: Why? No multiple connection at once?

Comment (andy89923) [reply internal/sbi/consumer/nrf_service.go:24]: I know there would not be multiple connections at once. But I think it's better to have the same structure/logic as others.

Comment (andy89923) [reply internal/sbi/consumer/nrf_service.go:24]: Or add a comment  to indicate that there would be no multiple-connection at once.

**Comment by andy89923** [comment]
Overall, LGTM🧑🏻‍💻

File: internal/sbi/consumer/nrf_service.go (hunk 7 above)
Comment (yccodr) [reply internal/sbi/consumer/nrf_service.go:24]: Comment added

File: internal/sbi/api_nsselection.go [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/free5gc/nssf/internal/logger"
+)
+
+func (s *Server) getNsSelectionRoutes() []Route {
+	return []Route{
+		{
+			"Helth Check",
```
Comment (blackcat-118) [inline internal/sbi/api_nsselection.go]: misspelling "Health" :)

**Review by andy89923 — APPROVED** [review]

**Comment by blackcat-118** [comment]
LGTM.

**Review by ianchen0119 — APPROVED** [review]
LGTM.

**Pull request merged**